src/Network/AWS/ARN/Internal/Lens.hs

```diff
@@ -16,33 +16,45 @@ module Network.AWS.ARN.Internal.Lens where
 import Data.Functor.Const (Const (..))
 import Data.Functor.Identity (Identity (..))
 import Data.Monoid (First (..))
-import Data.Profunctor (Profunctor(..))
+import Data.Profunctor (Profunctor (..))
 import Data.Profunctor.Choice (Choice (..))
 import Data.Tagged (Tagged (..))
 
 type Lens' s a = forall f. Functor f => (a -> f a) -> s -> f s
 
+type Getting r s a = (a -> Const r a) -> s -> Const r s
+
 type Setter s a = (a -> Identity a) -> s -> Identity s
 
 set :: Setter s a -> a -> s -> s
 set l = over l . const
 {-# INLINE set #-}
 
+(.~) :: Setter s a -> a -> s -> s
+(.~) = set
+
+infixr 4 .~
+
 over :: Setter s a -> (a -> a) -> s -> s
 over l f = runIdentity . l (Identity . f)
 {-# INLINE over #-}
 
+(^.) :: s -> Getting a s a -> a
+s ^. l = getConst $ l Const s
+
+infixl 8 ^.
+
 type Prism' s a =
   forall p f. (Choice p, Applicative f) => p a (f a) -> p s (f s)
 
 prism' :: forall s a. (a -> s) -> (s -> Maybe a) -> Prism' s a
-prism' to from p = dimap from' to' $ right' p
+prism' inj prj p = dimap prj' inj' $ right' p
   where
-    to' :: Applicative f => Either s (f a) -> f s
-    to' = either pure (fmap to)
+    inj' :: Applicative f => Either s (f a) -> f s
+    inj' = either pure (fmap inj)
 
-    from' :: s -> Either s a
-    from' s = maybe (Left s) Right $ from s
+    prj' :: s -> Either s a
+    prj' s = maybe (Left s) Right $ prj s
```

Comment:
I'm assuming this is to be consistent with `lens`?

Reply:
It's to avoid name shadowing with `from`, which is now a top-level definition.